src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -1450,8 +1450,11 @@ bool getHPWaterHeaterInput(EnergyPlusData &state)
             HPWH.DXCoilTypeNum = 0;
             if (HPWH.bIsIHP) {
                 HPWH.DXCoilType = "COILSYSTEM:INTEGRATEDHEATPUMP:AIRSOURCE";
+                int VSCoilNum = state.dataIntegratedHP->IntegratedHeatPumps(HPWH.DXCoilNum).SCWHCoilIndex;
+                HPWH.dxCoilAvailSched = state.dataVariableSpeedCoils->VarSpeedCoil(VSCoilNum).availSched;
             } else {
                 HPWH.DXCoilType = state.dataVariableSpeedCoils->VarSpeedCoil(HPWH.DXCoilNum).VarSpeedCoilType;
+                HPWH.dxCoilAvailSched = state.dataVariableSpeedCoils->VarSpeedCoil(HPWH.DXCoilNum).availSched;
             }
```

Comment:
Set a pointer for the availability schedule of the DX Coil in the HPWH during the input, then apply the schedule irrespective of the DX coil type.